File: experiments/polyglot/services/java/test/polyglot/IdentityServiceTest.java
package polyglot;

import dev.drim.azimuth.Azimuth;

public final class IdentityServiceTest {
    private IdentityServiceTest() {}

    @Azimuth.Covers(
            spec = "polyglot/identity",
            scenario = "java-identifies",
            scope = Azimuth.Scope.unit,
            quantification = Azimuth.Quantification.example,
            oracle = Azimuth.Oracle.direct)
    public static void main(String[] arguments) {
        if (!IdentityService.identity().equals("java")) {
            throw new AssertionError("Java identity changed");
        }
    }
}

File: experiments/polyglot/services/kotlin/src/test/kotlin/polyglot/IdentityServiceTest.kt
package polyglot

import dev.drim.azimuth.Azimuth
import kotlin.test.Test
import kotlin.test.assertEquals

class IdentityServiceTest {
    @Test
    @Azimuth.Covers(
        spec = "polyglot/identity",
        scenario = "kotlin-identifies",
        scope = Azimuth.Scope.unit,
        quantification = Azimuth.Quantification.example,
        oracle = Azimuth.Oracle.direct,
    )
    fun identityIsKotlin() {
        assertEquals("kotlin", IdentityService.identity())
    }
}
